### PR TITLE
test(core): add api response guard and error case tests to connector api

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

add api response guard and error case tests to connector api

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] docs

OR

- [ ] This PR is not applicable for the checklist

### CODE REVIEW

**Review by darcyYe — APPROVED** [review]
Overall LGTM, should we ignore response status 401?

File: packages/core/src/routes/connector/index.ts [hunk 1]
```diff
@@ -156,6 +163,8 @@ export default function connectorRoutes<T extends AuthedRouter>(
       query: object({
         target: string().optional(),
       }),
+      response: connectorResponseGuard.array(),
+      status: [200, 400],
```
Comment (charIeszhao) [inline packages/core/src/routes/connector/index.ts:167]: 200 and 404?

File: packages/core/src/routes/connector/index.ts [hunk 2]
```diff
@@ -251,6 +264,8 @@ export default function connectorRoutes<T extends AuthedRouter>(
       body: Connectors.createGuard
         .pick({ config: true, metadata: true, syncProfile: true })
         .partial(),
+      response: connectorResponseGuard,
+      status: [200, 400, 404, 422],
```
Comment (charIeszhao) [inline packages/core/src/routes/connector/index.ts:268]: Do we throw 400 in the code?

File: packages/core/src/routes/connector/index.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/connector/index.ts:167]: updated

File: packages/core/src/routes/connector/index.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/connector/index.ts:268]: Yes, there are several "assetThat", they will throw 400 by default.

File: packages/core/src/routes/connector/index.ts (hunk 1 above)
Comment (charIeszhao) [reply packages/core/src/routes/connector/index.ts:167]: Forget to push the code?

**Pull request merged**